lib/ansible/modules/web_infrastructure/ansible_tower/tower_organization.py

```diff
@@ -0,0 +1,141 @@
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: tower_organization
+version_added: "2.3"
+short_description: create, update, or destroy Ansible Tower organizations
+description:
+    - Create, update, or destroy Ansible Tower organizations. See
+      U(https://www.ansible.com/tower) for an overview.
+options:
+    name:
+      description:
+        - Name string to use for the organization.
+      required: True
+      default: null
+    description:
+      description:
+        - String to use for the organization.
+      required: False
+      default: null
+    state:
+      description:
+        - Desired state of the resource.
+      required: False
+      default: "present"
+      choices: ["present", "absent"]
+    config_file:
+      description:
+        - Path to the Tower config file. See notes.
+      required: False
+      default: null
+
+
+requirements:
+  - "python >= 2.6"
+  - "ansible-tower-cli >= 3.0.3"
+
+notes:
+  - If no I(config_file) is provided we will attempt to use the tower-cli library
+    defaults to find your Tower host information.
+  - I(config_file) should contain Tower configuration in the following format:
+      host=hostname
+      username=username
+      password=password
+'''
+
+
+EXAMPLES = '''
+  - tasks
+    - name: Create organization
+      tower_organization:
+          name: "Foo"
+          description: "Foo bar organization"
+          state: present
+          config_file: "~/tower_cli.cfg"
+'''
+
+import os
+
+try:
+    import tower_cli
+    import tower_cli.utils.exceptions as exc
+    from tower_cli.utils import parser
+    from tower_cli.conf import settings
+
+    HAS_TOWER_CLI = True
+except ImportError:
+    HAS_TOWER_CLI = False
+
+
+def tower_auth_config(module):
+    config_file = module.params.get('config_file')
+    if not config_file:
+        return {}
+
+    config_file = os.path.expanduser(config_file)
+    if not os.path.exists(config_file):
+        module.fail_json(msg='file not found: %s' % config_file)
+    if os.path.isdir(config_file):
+        module.fail_json(msg='directory can not be used as config file: %s' % config_file)
+
+    with open(config_file, 'rb') as f:
+        return parser.string_to_dict(f.read())
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec = dict(
+            name = dict(),
+            description = dict(),
+            config_file = dict(),
+            state = dict(choices=['present', 'absent'], default='present'),
+        ),
+        supports_check_mode=False
```

Comment:
Sending a query to tower and attempting to guess at the outcome based on the response would be just as effective as writing ... Occasionally you would get some extra detail, but generally you'd just be making a WAG.

    if module.check_mode:
        module.json_exit(changed=True